server/src/test/python/cromwell-az-e2e-test/cromwell_az_e2e_test/az_e2e.py

```
@@ -0,0 +1,173 @@
+#                 print(f"{app} is still provisioning")
+#                 break
+#             print(f"App status: {entries['status']}")
+#             app_url = entries['proxyUrls'][app]
+#             break 
+
+#     if app_url is None: 
+#         print(f"{app} is missing in current workspace")
+#     else:
+#         print(f"{app} url: {app_url}")
+
+#     return app_url
+
+def submit_workflow_to_cromwell(app_url, workflow_test_name):
+    absolute_file_path = os.path.dirname(__file__)
+    workflow_source_path = os.path.join(absolute_file_path, 'workflow_files/hello.wdl')
+    workflow_inputs_path = os.path.join(absolute_file_path, 'workflow_files/hello.inputs')
+    workflow_endpoint = f'{app_url}/api/workflows/v1'
+    headers = {"Authorization": f'Bearer {bearer_token}',
+              "accept": "application/json",
+    }
+    files = {
+        'workflowSource': ('hello.wdl', open(workflow_source_path, 'rb'), 'application/octet-stream'),
+        'workflowInputs': ('hello.inputs', open(workflow_inputs_path, 'rb'), 'application/octet-stream'),
+        'workflowType': 'WDL',
+        'workflowTypeVersion': '1.0'
+    }
+    response = requests.post(workflow_endpoint, headers=headers, files=files)
+    handle_failed_request(response, f"Error submitting workflow to Cromwell for {workflow_test_name}", 201)
+    output_message(response.json())
+    return response.json()
+
+def get_workflow_information(app_url, workflow_id):
+    workflow_endpoint = f'{app_url}/api/workflows/v1/{workflow_id}/metadata'
+    headers = {"Authorization": f'Bearer {bearer_token}',
+              "accept": "application/json"}
+    output_message(f"Fetching workflow metadata for {workflow_id}")
+    response = requests.get(workflow_endpoint, headers=headers)
+    handle_failed_request(response, f"Error fetching workflow metadata for {workflow_id}")
+    output_message(response.json())
+    return response.json()
+
+def get_completed_workflow(app_url, workflow_ids, max_retries=4):
```

I can see the combination of `max_retries` and the amount of time to wait between checks getting confusing. Might be better to make both into function arguments, or just let the test fail with an informative error when our token expires... IF we are confident Janitor will clean things up for us.